swaybar/tray/watcher.c

```diff
@@ -33,18 +29,23 @@ static int handle_lost_service(sd_bus_message *msg,
 
 	if (!*new_owner) {
 		struct swaybar_watcher *watcher = data;
-		int idx = list_seq_find(watcher->items,
-				using_standard_protocol(watcher) ? cmp_id : cmp_service, service);
-		if (idx != -1) {
+		for (int idx = 0; idx < watcher->items->length; ++idx) {
 			char *id = watcher->items->items[idx];
-			sway_log(SWAY_DEBUG, "Unregistering Status Notifier Item '%s'", id);
-			list_del(watcher->items, idx);
-			sd_bus_emit_signal(watcher->bus, obj_path, watcher->interface,
-					"StatusNotifierItemUnregistered", "s", id);
-			free(id);
+			sway_log(SWAY_DEBUG, "DEBUG %s %s", id, service);
+			if ((using_standard_protocol(watcher) ? cmp_id(id, service) :
```

Comment:
IMO embedding a ternary in a condition is a bit nasty. I would recommend declaring a bool.